napari/_qt/_qapp_model/qactions/_layer.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import annotations
+
+import json
+import pickle
+
+import numpy as np
+from app_model.expressions import parse_expression
+from app_model.types import Action
+from qtpy.QtCore import QMimeData
+from qtpy.QtWidgets import QApplication
+
+from napari._app_model.constants import MenuGroup, MenuId
+from napari.components import LayerList
+from napari.layers import Layer
+from napari.utils.notifications import show_warning
+from napari.utils.translations import trans
+
+__all__ = ('Q_LAYER_ACTIONS', 'is_valid_spatial_in_clipboard')
+
+
+def _numpy_to_list(d: dict) -> dict:
+    for k, v in list(d.items()):
+        if isinstance(v, np.ndarray):
+            d[k] = v.tolist()
+    return d
+
+
+def _set_data_in_clipboard(data: dict) -> None:
+    data = _numpy_to_list(data)
+    clip = QApplication.clipboard()
+    if clip is None:
+        show_warning('Cannot access clipboard')
+        return
+
+    d = json.dumps(data)
+    p = pickle.dumps(data)
+    mime_data = QMimeData()
+    mime_data.setText(d)
+    mime_data.setData('application/octet-stream', p)
+
+    clip.setMimeData(mime_data)
+
+
+def _copy_spatial_to_clipboard(layer: Layer) -> None:
+    _set_data_in_clipboard(
+        {
+            'affine': layer.affine.affine_matrix,
+            'rotate': layer.rotate,
+            'scale': layer.scale,
+            'shear': layer.shear,
+            'translate': layer.translate,
+        }
+    )
+
+
+def _copy_affine_to_clipboard(layer: Layer) -> None:
+    _set_data_in_clipboard({'affine': layer.affine.affine_matrix})
+
+
+def _copy_rotate_to_clipboard(layer: Layer) -> None:
+    _set_data_in_clipboard({'rotate': layer.rotate})
+
+
+def _copy_shear_to_clipboard(layer: Layer) -> None:
+    _set_data_in_clipboard({'shear': layer.shear})
+
+
+def _copy_scale_to_clipboard(layer: Layer) -> None:
+    _set_data_in_clipboard({'scale': layer.scale})
+
+
+def _copy_translate_to_clipboard(layer: Layer) -> None:
+    _set_data_in_clipboard({'translate': layer.translate})
+
+
+def _get_spatial_from_clipboard() -> dict | None:
+    clip = QApplication.clipboard()
+    if clip is None:
+        return None
+
+    mime_data = clip.mimeData()
+    if mime_data is None:
+        return None
+    if mime_data.data('application/octet-stream'):
+        return pickle.loads(mime_data.data('application/octet-stream'))  # type: ignore[arg-type]
+
+    return json.loads(mime_data.text())
+
+
+def _paste_spatial_from_clipboard(ll: LayerList) -> None:
+    try:
+        loaded = _get_spatial_from_clipboard()
+    except (json.JSONDecodeError, pickle.UnpicklingError):
+        show_warning('Cannot parse clipboard data')
+        return
+    if loaded is None:
+        show_warning('Cannot access clipboard')
+        return
+
+    for layer in ll.selection:
+        for key in loaded:
+            data = loaded[key]
```

Comment:
This confused me at first. I g` 
```suggestion
            payload = loaded[key]
```
`data` confused me, because I got layer data mentally. This is just the clipboard data.
maybe payload is better? or something else.
`key_data`?
`clipboard_data`?
I dunno

Reply:
I have changed variable name to `loaded_attr_value`. Is it ok?